Yoast/YoastSEO.js
# Relevant words filters review

## Summary

This PR can be summarized in the following changelog entry: Improve filtering for Internal Linking Suggestions and Insights for Italian, Spanish, Dutch, French, English and German.

## General
- Split numerals into cardinal and ordinal numbers. Cardinal numbers are filtered at the beginning and ending of combinations, while ordinal numbers are only filtered at the ending.
- Remove the 1-syllable filter. Fixes #1166.
- Filter relative pronouns also at the ending. Related #1102.
- Filter words from the miscellaneous lists also at the beginning.
- Add filters for beginning and ending for: indefinitePronouns, locativeAdverbs, adverbialGenitives, prepositionalAdverbs, recipeWords
- Filter words from the intensifiers lists at the beginning and ending. Fixes #1110.
- Adapt tests to new filters.

## English
- Add a continuousVerbs category. These words are filtered only at the end. 
- All other verbs are filtered at the beginning and ending. 
- Remove passiveAuxiliaries filter, because these are already included in the verbs filter.
- Filter generalAdjectivesAdverbs at ending.

## Dutch
- Filter generalAdjectivesAdverbs at ending.
- Move infinitive verbs to separate infinitive categories. Infinitives are only filtered at the beginning. Other verbs at the beginning and ending.
- Filter timeWords at beginning and ending.
- Filter reciprocal pronouns at the ending.
- Change the name ‘interrogativeAdverbs’ to ‘interrogativeProAdverbs’ for consistency with other languages.
- Add correlative conjunctions to the conjunctions filter.
- Rename beginningVerbs export category to infinitives.

## French
- Split generalAdjectivesAdverbs into generalAdjectivesAdverbsPreceding and generalAdjectivesAdverbs. generalAdjectivesAdverbsPreceding are filtered at the ending, while generalAdjectivesAdverbs are filtered at the beginning.
- Remove ‘premier’ and ‘première’ from the ordinal numerals, because these are also nouns (prime minister and first showing of a film). 
- Filter interrogatives at beginning and ending.
- Add delexicalizedVerbs to the verbs filter.
- Rename beginningVerbs export category to infinitives.

## German
- Filter generalAdjectivesAdverbs at ending.
- Remove reflexive pronouns that were already included in personalPronounsAccusative.
- Added delexicalizedVerbs to the verbs category.
- Filter timeWords at beginning and ending.
- Add reciprocal pronouns. These are filtered at the ending.
- Rename beginningVerbs export category to infinitives.

## Italian
- Split generalAdjectivesAdverbs into generalAdjectivesAdverbsPreceding and generalAdjectivesAdverbs. generalAdjectivesAdverbsPreceding are filtered at the ending, while generalAdjectivesAdverbs are filtered at the beginning.
- Remove reflexivePronouns category and move those words to generalAdjectivesAdverbs because they also mean 'same' and aren't used in combination with infinitives in their reflexive meaning.
- Remove interrogative export category, because all those word categories were also included in the relativePronouns export category.
- Remove unused passiveAuxiliary export category.
- Remove infinitivePassiveAuxiliaries from the verbs filter.
- Rename beginningVerbs export category to infinitives.
- Rename beginningVerbs export category to infinitives.

## Spanish
- Filter generalAdjectivesAdverbs at the beginning.
- Remove reflexivePronoun category, because the only word in that category was already included in the personalPronounsAccusatives.
- Add personalPronounsComitative, personalPronounsPrepositional and personalPronounsGenitive to the `all` export.
- Add correlative conjunctions to the `conjunctions` export

## Test instructions

This PR can be tested by following these steps:
* Use the browserified relevant words example.
* Write a text in English, German, French, Dutch, Spanish or Italian.
* Set the locale to en_US, de_DE, fr_FR, nl_NL, es_ES, or it_IT.
*  Check whether you see strange word combinations. For the languages you don't speak, switch between locales and make sure the suggestions change. 

Fixes #1179
